### PR TITLE
feat: evaluator enhancements

### PR DESCRIPTION
**Pull request by axiomofjoy**
- adds scores to evaluators, so that evaluators now capture the directionality of the evaluation (e.g., relevant = 1, irrelevant = 0)
- fixes bug where `run_evals` would output nans for explanations when LLM output could not be parse
- removes `from_criteria` as a user-facing concept on `LLMEvaluator`
- adds unit tests
- adds evaluators for all the major criteria (e.g., `RelevanceEvaluator`, `HallucinationEvaluator`, etc.)
- adds evals quickstart notebook

### CODE REVIEW

File: src/phoenix/experimental/evals/evaluators.py [hunk 1]
```diff
@@ -149,29 +150,98 @@ async def aevaluate(
...
+            "ReferenceLinkCorrectnessEvaluator",
+            EvalCriteria.REFERENCE_LINK_CORRECTNESS.value,
+            'Leverages an LLM to evaluate whether a piece of documentation (stored under a "reference" column) answers a customer\'s query (stored under an "input" column).',  # noqa: E501
+        ),
+        (
+            "HumanVsAIEvaluator",
+            EvalCriteria.HUMAN_VS_AI.value,
+            'Leverages an LLM to evaluate whether an AI-generated answer (stored under an "ai_generated_answer" column) matches the content of a human reference answer (stored under a "correct_answer" column) given a query (stored under a "question" column).',  # noqa: E501
+        ),
+    ),
```
Comment (mikeldking) [inline src/phoenix/experimental/evals/evaluators.py:243]: I would probably nix these

File: src/phoenix/experimental/evals/functions/classify.py [hunk 2]
```diff
@@ -380,23 +381,23 @@ def run_evals(
 
     async def _arun_eval(
         payload: RunEvalsPayload,
-    ) -> Tuple[EvaluatorIndex, RowIndex, Label, Explanation]:
-        label, explanation = await payload.evaluator.aevaluate(
+    ) -> Tuple[EvaluatorIndex, RowIndex, Label, Score, Explanation]:
+        label, score, explanation = await payload.evaluator.aevaluate(
```
Comment (mikeldking) [inline src/phoenix/experimental/evals/functions/classify.py:385]: Kinda confused why classify has to be touched. Can't it be done independently?

Comment (axiomofjoy) [reply src/phoenix/experimental/evals/functions/classify.py:385]: This is a tweak to `run_evals`, which currently lives inside the `classify` sub-module.

File: src/phoenix/experimental/evals/evaluators.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/experimental/evals/evaluators.py]: addressed in be996d61

File: src/phoenix/experimental/evals/functions/classify.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/experimental/evals/functions/classify.py:392]: I see, nice.

**Review by mikeldking — APPROVED** [review]

**Pull request merged**